File: app/src/main/java/br/com/medeve/Controlers/SplashScreen.java
package br.com.medeve.Controlers;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.VersionedPackage;
import android.os.Build;
import android.os.Bundle;
import android.os.Handler;
import android.util.Log;
import android.widget.TextView;

import br.com.medeve.R;

public class SplashScreen extends AppCompatActivity {

    TextView txt_versao;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        txt_versao = findViewById(R.id.versão_app);

        Handler h_splash = new Handler();

        h_splash.postDelayed(new Runnable() {
            @Override
            public void run() {

                Intent i_splash = new Intent(SplashScreen.this, EntrarUsuario.class);
                startActivity(i_splash);
                finish();

            }
        }, 2000);

        try {
            String version = this.getPackageManager().getPackageInfo("br.com.medeve", 0).versionName;
            txt_versao.setText(version);
        } catch (Exception e) {

            Log.d("Erro", "Versão não encontrada", e);
        }
    }
}

File: app/src/main/java/br/com/medeve/Activitys/SplashScreen.kt
package br.com.medeve.Activitys

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import br.com.medeve.R
import kotlinx.android.synthetic.main.activity_splash_screen.*

class SplashScreen : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)

        val h_splash = Handler()

        h_splash.postDelayed({
            val i_splash = Intent(this@SplashScreen, EntrarUsuarioActView::class.java)
            startActivity(i_splash)
            finish()
        }, 2000)

        try {
            val version = this.packageManager.getPackageInfo("br.com.medeve", 0).versionName
            versao_app.text = version
        } catch (e: Exception) {
            Log.d("Erro", "Versão não encontrada", e)
        }

    }
}